### PR TITLE
refactor(lib): simplify some method

### PR DESCRIPTION
**Pull request by VictorKmin**
change if(){condition} else{} to condition ? true : false

love you

### Pull Request check-list

_Please make sure to review and check all of these items:_

- [ ] Does `npm run test` or `npm run test-DIALECT` pass with this change (including linting)?
- [ ] Does the description below contain a link to an existing issue (Closes #[issue]) or a description of the issue you are solving?
- [ ] Have you added new tests to prevent regressions?
- [ ] Is a documentation update included (if this change modifies existing APIs, or introduces new ones)?
- [ ] Did you update the typescript typings accordingly (if applicable)?
- [ ] Did you follow the commit message conventions explained in [CONTRIBUTING.md](https://github.com/sequelize/sequelize/blob/master/CONTRIBUTING.md)?

### CODE REVIEW

File: lib/associations/has-many.js [hunk 1]
```diff
@@ -212,11 +212,9 @@ class HasMany extends Association {
       where;
 
     if (Object.prototype.hasOwnProperty.call(options, 'scope')) {
-      if (!options.scope) {
-        Model = Model.unscoped();
-      } else {
-        Model = Model.scope(options.scope);
-      }
+      Model = !options.scope ?
```
Comment (Americas) [inline lib/associations/has-many.js]: WDYT about switching the statements and removing the negation?

File: lib/sequelize.js [hunk 2]
```diff
@@ -1232,8 +1233,8 @@ class Sequelize {
     if (Object.prototype.hasOwnProperty.call(attribute, 'defaultValue')) {
       if (typeof attribute.defaultValue === 'function' && (
         attribute.defaultValue === DataTypes.NOW ||
-          attribute.defaultValue === DataTypes.UUIDV1 ||
-          attribute.defaultValue === DataTypes.UUIDV4
+        attribute.defaultValue === DataTypes.UUIDV1 ||
+        attribute.defaultValue === DataTypes.UUIDV4
```
Comment (Americas) [inline lib/sequelize.js]:
```diff
-        attribute.defaultValue === DataTypes.NOW ||
-        attribute.defaultValue === DataTypes.UUIDV1 ||
-        attribute.defaultValue === DataTypes.UUIDV4
+        [DataTypes.NOW, DataTypes.UUIDV1, DataTypes.UUIDV4].includes(attribute.defaultValue)
```

**Comment by papb** [comment]
Hello! I see you are a first-time contributor, thank you for taking the time to help Sequelize! I hope to see more PRs from you in the future!

I will review your PR now

File: lib/data-types.js [hunk 3]
```diff
@@ -441,16 +441,19 @@ class DATE extends ABSTRACT {
     return value;
   }
   _isChanged(value, originalValue) {
```
Comment (papb) [inline lib/data-types.js]: I like that you wanted to refactor this `_isChanged` method, however I think an even better refactor can be done:

```js
_isChanged(value, originalValue) {
  if (value instanceof Date && originalValue instanceof Date) {
    return value.getTime() !== originalValue.getTime();
  }
  return value !== originalValue;
}
```

Please convince yourself that my code above is equivalent to the current code and use it instead :)

File: lib/data-types.js [hunk 4]
```diff
@@ -488,11 +491,8 @@ class DATEONLY extends ABSTRACT {
     if (originalValue && !!value && originalValue === value) {
       return false;
     }
-    // not changed when set to same empty value
-    if (!originalValue && !value && originalValue === value) {
-      return false;
-    }
-    return true;
+
+    return !(!originalValue && !value && originalValue === value);
```
Comment (papb) [inline lib/data-types.js]: Same as above, please use my improved refactor for the whole function.

File: lib/query-interface.js [hunk 5]
```diff
@@ -214,16 +213,12 @@ class QueryInterface {
...
-    }
+    const promise = this.sequelize.options.dialect === 'postgres' ?
+      PostgresQueryInterface.ensureEnums(this, tableName, attributes, options, model) :
+      Promise.resolve();
 
-    if (
-      !tableName.schema &&
-      (options.schema || !!model && model._schema)
-    ) {
+    const canAddSchema = !tableName.schema && (options.schema || !!model && model._schema);
```
Comment (papb) [inline lib/query-interface.js]: The `!!` boolean cast is unnecessary, also I know it's debatable but I think adding parentheses around the `&&` improves readability; not everyone knows that `&&` has higher precedence than `||` in JS.

```suggestion
    const canAddSchema = !tableName.schema && (options.schema || (model && model._schema));
```

File: lib/sequelize.js [hunk 6]
```diff
@@ -1109,7 +1109,8 @@ class Sequelize {
           // Rollback transaction if not already finished (commit, rollback, etc)
           // and reject with original error (ignore any error in rollback)
           return Promise.try(() => {
-            if (!transaction.finished) return transaction.rollback().catch(() => {});
+            if (!transaction.finished) return transaction.rollback().catch(() => {
+            });
```
Comment (papb) [inline lib/sequelize.js]: I think the previous format was better here, please revert

File: lib/sequelize.js (hunk 2 above)
Comment (papb) [reply lib/sequelize.js]: I agree with @Americas here as well :+1:

File: lib/sequelize.js (hunk 6 above)
Comment (VictorKmin) [reply lib/sequelize.js]: My IDE do this automatically. I will revert. Sorry about that.

**Comment by WikiRik** [comment]
Sorry for not merging this earlier, but I will close this since much has changed since October 2019.

**Pull request closed without merging**